### ISSUE TITLE
A question about the loss function

### ISSUE THREAD

**Issue by LostStars3** [thangvubk/Cascade-RPN#4]
Hi,
It is mentioned in the paper that the use of anchor-free and sample discrimination in stage 1, an anchor is a positive sample if its center is inside the center region of an object.
And then, I found the classification loss function that you use at this stage is CrossEntropyLoss. I wonder why you didn't use focal loss like the other anchor-free methods. Wouldn't you encounter the problem of unbalanced positive and negative samples.
Looking forward to your reply, thank you!

**Comment by thangvubk** [comment]
The class head is used for only the last stage. (See Figure 2.e in the paper and check [here](https://github.com/thangvubk/Cascade-RPN/blob/462797464bd7c2242201b488b5695e043693bad4/configs/cascade_rpn/cascade_rpn_r50_fpn_1x.py#L29)). It is for two reasons:
1. Reduce overhead
2. Avoid imbalance at the first stage because the stage 2 has more good proposals than stage 1 to some extent.

You can also try to use focal loss as classification loss of stage 2. It is expected that you may get some gain. Here, we just follow standard RPN that use CrossEntropy for classification loss.

**Comment by LostStars3** [comment]
> The class head is used for only the last stage. (See Figure 2.e in the paper and check [here](https://github.com/thangvubk/Cascade-RPN/blob/462797464bd7c2242201b488b5695e043693bad4/configs/cascade_rpn/cascade_rpn_r50_fpn_1x.py#L29)). It is for two reasons:
> 
> 1. Reduce overhead
> 2. Avoid imbalance at the first stage because the stage 2 has more good proposals than stage 1 to some extent.
> 
> You can also try to use focal loss as classification loss of stage 2. It is expected that you may get some gain. Here, we just follow standard RPN that use CrossEntropy for classification loss.

I see, thank you for your reply!
